lib/route/src/Obelisk/Route.hs

```diff
@@ -433,18 +435,23 @@ instance (Applicative check, Monad parse) => Monoidal (Encoder check parse) (,)
 -- Specific instances of encoders
 --------------------------------------------------------------------------------
 
+{-# DEPRECATED isoEncoder "Use viewEncoder instead" #-}
 -- | Given a valid 'Iso' from lens, construct an 'Encoder'
 isoEncoder :: (Applicative check, Applicative parse) => Iso' a b -> Encoder check parse a b
-isoEncoder f = unsafeMkEncoder $ EncoderImpl
-  { _encoderImpl_encode = view f
-  , _encoderImpl_decode = pure . view (from f)
+isoEncoder p = viewEncoder (from p)
+
+-- | Given a valid 'Iso' from lens, construct an 'Encoder'
+viewEncoder :: (Applicative check, Applicative parse) => Iso' a b -> Encoder check parse b a
```

Comment:
Why is this contravariant?  [`view`](https://hackage.haskell.org/package/lens-4.19.2/docs/Control-Lens-Getter.html#v:view) is covariant.